streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -527,35 +531,44 @@ private ApplicationState buildApplicationState(final TopologyMetadata topologyMe
                                               + "tasks for source topics vs changelog topics.");
         }
 
-        final Set<TaskId> logicalTaskIds = unmodifiableSet(sourcePartitionsForTask.keySet());
-        final Set<DefaultTaskTopicPartition> allTopicPartitions = new HashSet<>();
+        final Set<DefaultTaskTopicPartition> topicsRequiringRackInfo = new HashSet<>();
+        final AtomicBoolean rackInformationFetched = new AtomicBoolean(false);
+        final Runnable fetchRackInformation = () -> {
+            if (!rackInformationFetched.get()) {
+                RackUtils.annotateTopicPartitionsWithRackInfo(cluster,
```

Comment:
very small nit (tack onto any followup PR): weird line break, either keep everything on one line or move the `cluster` variable to the 2nd line with the other params